transport/inbound.go

```diff
@@ -46,3 +46,17 @@ type Inbound interface {
 	// Implementations can assume that this function is called at most once.
 	Start(handler Handler, deps Deps) error
 }
+
+// OnewayInbound is a transport that knows how to receive oneway requests
+// for procedure calls.
+type OnewayInbound interface {
+	BaseInbound
+
+	// Starts accepting new requests and dispatches them to the given Handler.
+	//
+	// The function MUST return immediately, although it SHOULD block until
+	// the inbound is ready to start accepting new requests.
+	//
+	// Implementations can assume that this function is called at most once.
+	Start(handler OnewayHandler, deps Deps) error
```

Comment:
Yes, so I'll essentially add a RegisterOneway function later on so the  registry can distinguish the handlers.
---
On another note: I think I'm going to have to rethink this part. Ideally we should be able to start a transport that understands unary AND oneway requests (I'm thinking of the TChannel/http case). So we should be able to give an outbound the handlers it supports and then start taking requests on either of those handlers